src/test/java/org/sonarlint/intellij/analysis/SonarLintTaskTest.java

```diff
@@ -108,7 +116,133 @@ public void testTask() {
       .extracting("implementationClass")
       .contains("org.sonarlint.intellij.editor.SonarExternalAnnotator");
     verifyNoMoreInteractions(sonarLintAnalyzer);
-    verifyNoMoreInteractions(processor);
+    verifyNoMoreInteractions(liveIssueBuilder);
+  }
+
+  @Test
+  public void shouldIgnoreProjectLevelIssues() {
+    TaskListener listener = mock(TaskListener.class);
+    getProject().getMessageBus().connect(getProject()).subscribe(TaskListener.SONARLINT_TASK_TOPIC, listener);
+    when(sonarLintAnalyzer.analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class)))
+      .thenAnswer((Answer<AnalysisResults>) invocation -> {
+        IssueListener issueListener = invocation.getArgument(2);
+        Issue issue = SonarLintTestUtils.createIssue(1);
+        issueListener.handle(issue);
+        return analysisResults;
+      });
+
+    task.run(progress);
+
+    verify(sonarLintAnalyzer).analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class));
+    verify(issueManagerMock, never()).insertNewIssue(any(), any());
+    verifyNoMoreInteractions(liveIssueBuilder);
+  }
+
+  @Test
+  public void shouldIgnoreInvalidFiles() {
+    TaskListener listener = mock(TaskListener.class);
+    getProject().getMessageBus().connect(getProject()).subscribe(TaskListener.SONARLINT_TASK_TOPIC, listener);
+    when(sonarLintAnalyzer.analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class)))
+      .thenAnswer((Answer<AnalysisResults>) invocation -> {
+        IssueListener issueListener = invocation.getArgument(2);
+        Issue issue = mock(Issue.class);
+        ClientInputFile clientInputFile = mock(ClientInputFile.class);
+        when(issue.getInputFile()).thenReturn(clientInputFile);
+        when(clientInputFile.getClientObject()).thenReturn(mock(VirtualFile.class));
```

Comment:
Even if this is the default of Mockito to return `false` for methods returning `boolean`, I would find it more explicit and self documentary to do something like:
```suggestion
        VirtualFile vFile = mock(VirtualFile.class);
        when(vFile.isValid()).thenReturn(false);
        when(clientInputFile.getClientObject()).thenReturn(vFile);
```